scripts/MTweede_adapted_solution.R

```diff
@@ -0,0 +1,145 @@
+# Assignment 1:  
+library(tweedie) 
+library(ggplot2)
+library(doParallel)
+library(foreach)
+
+
+simTweedieTest <-  
+  function(N){ 
+    t.test( 
+      rtweedie(N, mu=10000, phi=100, power=1.9), 
+      mu=10000 
+    )$p.value 
+  } 
+
+
+# Assignment 2:
+MTweedieTests <-  
```

Comment:
I am not entirely sure what is wrong here, but here is one way you could have created a function to accomplish this:
```
function(N, M, sig, maxcores) {
    Cores <- min(parallel::detectCores(), maxcores)
    
    cl <- makeCluster(Cores)
    registerDoParallel(cl)
    
    m_per_core <- round(M / Cores)
    m_total <- m_per_core * Cores
    
    rejects_per_core <-
      foreach(
        i = 1:Cores,
        .combine = 'rbind',
        .packages = c('magrittr', 'dplyr', 'tweedie'),
        .export = c('simTweedieTest')
      ) %dopar%
      sum(replicate(m_per_core, simTweedieTest(N)) < sig)
    
    stopCluster(cl)
    
    sum(rejects_per_core) / m_total
  }
  ```
As you can see this divides the simulations evenly into the cores. Since there are likely more than 5 cores used, this would ensure that all of them are used. For the first parallell approach, it would only use 5 cores.